### ISSUE TITLE
"Side staking bot"

### ISSUE THREAD

**Issue by kremalicious** [oceanprotocol/market#1014]
The new funky thing preventing rug pulls, and not even sure where the name comes from but it's floating around in the team. Issue is for deciding if it needs some visualization with its own numbers on the Pool tab, and if so, how that would look like and then implement. Advantage of visualizing it is that UI helps in understanding what's special with the new pools, explaining it without a wall of text.

The facts:

- bot (another Ethereum account) for each pool auto-controls datatoken supply
- whenever someone stakes OCEAN into a pool, the pool mints datatokens
- just enough datatokens are minted to keep datatoken price stable (constant DT:OCEAN ratio)
- whenever OCEAN is unstaked, datatokens are burned
- bot will receive half of the liquidity provider's pool shares

That last point might mean we have to calculate/show things like creator/user liquidity differently and not base this all on the user's pool shares.

**Comment by trentmc** [comment]
A reference: Bancor has "single-sided staking". They have different underlying mechanics than Ocean, but the effect is the same: you're a liquidity provider into a pool yet you only have to give tokens on one side.

Here's what it looks like.
https://app.bancor.network/pools/add-liquidity/0xb1CD6e4153B2a390Cf00A6556b0fC1458C4A5533

In app, go to "Earn -> Pools". For a given token (e.g. ETH), see the "+" button. Click it.
![bancor0](https://user-images.githubusercontent.com/5305452/150537752-6c53ca06-ae42-4e50-af9f-d159c921f4e8.png)

After you've clicked, you get the following screen.
![bancor1](https://user-images.githubusercontent.com/5305452/150537754-ce887e90-38a5-4c5f-bf73-767318ac6dfa.png)

**Comment by trentmc** [comment]
Notes:

IMO we should _not_ call it "side staking bot" or "staking bot". We should call it "single sided staking" or "one-sided staking". It's much simpler for the user, because it's less about the "how" (robots? huh?) and more about the "what" (you only have to stake one side).

**Comment by kremalicious** [comment]
the contracts call it side staking, which then also populates to ocean.js calling it like that https://github.com/oceanprotocol/contracts/blob/v4main/contracts/pools/ssContracts/SideStaking.sol

**Comment by trentmc** [comment]
> the contracts call it "side staking", which then also populates to ocean.js calling it like that

"Side staking" is a terrible label for users; we definitely shouldn't use it in the GUI. 

More generally, just because a lower-level library uses a label X, it doesn't mean that it should be propagated to a GUI level. E.g. we don't expose `joinswapExternAmountIn()`. But you know that:)

There's also a good argument to rename away from "side staking" in the contracts and ocean.js, but I don't want to rock the boat.

**Comment by kremalicious** [comment]
yup agree, only dropped for reference cause I wondered in my description where it was coming from. I had so many questions about each of the 3 words :-)

**Comment by kremalicious** [comment]
- Your Liquidity, Pool Creator Statistics: leave only OCEAN liquidity, pool shares, move % share beside title
- opens up space to optionally have new section to visualize the bot
- use spot price everywhere, except for Use where we show the consume price
- pools are always 50/50 weight, hardcoded in contract

**Comment by kremalicious** [comment]
Checkpoint, over in #1057 I reduced the liquidity stats to their essence which is this:

<img width="468" alt="Screen Shot 2022-02-03 at 16 47 08" src="https://user-images.githubusercontent.com/90316/152388550-545cad3f-43e5-456a-a02a-7e5aecd9c77e.png">

Doing this by simply doubling the fetched ocean liquidity (because of hardcoded 50/50 weight), and ignoring fetched datatoken liquidity for those users.

Think already more clear for liquidity providers although already interesting effect shown. I added 10 OCEAN and right away getting shown I now only have 9.685 OCEAN which I assume is the `poolFee` kicking in on adding liquidity?
